components/logins/src/db.rs

```diff
@@ -419,6 +419,82 @@ impl LoginDb {
         Ok(login)
     }
 
+    pub fn import_multiple(&self, logins: &[Login]) -> Result<u64> {
+        let tx = self.unchecked_transaction()?;
+        let now_ms = util::system_time_ms_i64(SystemTime::now());
+        let sql = format!(
+            "INSERT OR IGNORE INTO loginsL (
+                hostname,
+                httpRealm,
+                formSubmitURL,
+                usernameField,
+                passwordField,
+                timesUsed,
+                username,
+                password,
+                guid,
+                timeCreated,
+                timeLastUsed,
+                timePasswordChanged,
+                local_modified,
+                is_deleted,
+                sync_status
+            ) VALUES (
+                :hostname,
+                :http_realm,
+                :form_submit_url,
+                :username_field,
+                :password_field,
+                :times_used,
+                :username,
+                :password,
+                :guid,
+                :time_created,
+                :time_last_used,
+                :time_password_changed,
+                :local_modified,
+                0, -- is_deleted
+                {new} -- sync_status
+            )",
+            new = SyncStatus::New as u8
+        );
+        let mut num_failed = 0;
+        for login in logins {
+            if login.check_valid().is_err() {
+                log::warn!("Skipping login {} as it is invalid.", login.guid);
+                continue;
+            }
+            let old_guid = &login.guid; // Keep the old GUID around so we can debug errors easily.
+            let guid = Guid::random(); // Override any provided GUID.
```

Comment:
Err, why? This seems likely to cause duplicates if the user ever starts using sync.

We should validate that their GUID is sane (the server requires BSO ids to be at most 64 characters, and only printable ASCII characters. We should also reject an empty guid because it's just asking for trouble), but use it if it's sane.